### PR TITLE
Bring up all dependencies when running a single service.

### LINKED ISSUE #2833: run not looking at depends_on
I think that `docker-compose run <service>` should bring up the dependencies of that service, the same as `docker-compose up <service>` would. There is already the `--no-deps` flag for run, which implies that the default behavior is to bring up dependencies.

If this is the intended behavior, then I am not seeing it. If I have a `docker-compose.yml` file like this:

```yaml
version: '2'
services:
  web:
    image: postgres
    command: sh -c 'psql -h db -p 5432 -U postgres -d postgres -c "\list"'
    entrypoint: /tmp/entrypoint.sh
    volumes:
      - ./entrypoint.sh:/tmp/entrypoint.sh
    depends_on:
       - db
  db:
    image: postgres
```

Then running `docker-compose run web` won't bring up the `db` container.

### PR DESCRIPTION
**Pull request by shin-**
Fixes #2833

### CODE REVIEW

File: tests/acceptance/cli_test.py [hunk 1]
```diff
@@ -738,6 +738,14 @@ def test_run_service_with_links(self):
         self.assertEqual(len(db.containers()), 1)
         self.assertEqual(len(console.containers()), 0)
 
+    def test_run_service_with_dependencies(self):
+        self.base_dir = 'tests/fixtures/v2-dependencies'
```
Comment (dnephin) [inline tests/acceptance/cli_test.py]: I think you forgot to `git add` this ?

Comment (shin-) [inline tests/acceptance/cli_test.py]: Yup, I had. Fixed!

File: tests/acceptance/cli_test.py [hunk 2]
```diff
@@ -738,6 +738,14 @@ def test_run_service_with_links(self):
         self.assertEqual(len(db.containers()), 1)
         self.assertEqual(len(console.containers()), 0)
 
+    def test_run_service_with_dependencies(self):
```
Comment (dnephin) [inline tests/acceptance/cli_test.py]: I think this needs the `@v2_only()` decorator

**Comment by dnephin** [comment]
LGTM

**Comment by saulshanabrook** [comment]
Thanks @dnephin this is great :+1:

**Pull request merged**